### PR TITLE
[release] Prepare release core-1.9.0

### PR DESCRIPTION
**Pull request by opentelemetrybot**
Note: This PR was opened automatically by the [prepare release workflow](https://github.com/open-telemetry/opentelemetry-dotnet/actions/workflows/prepare-release.yml).

## Changes

* CHANGELOG files updated for projects being released.
* Public API files updated for projects being released (only performed for stable releases).

### CODE REVIEW

**Comment by CodeBlanch** [comment]
@alanwest @vishweshbankwar @Kielek @utpilla @cijothomas

Release isn't going to happen for another couple days but please review public API on this diff and raise any concerns early!

**Review by vishweshbankwar — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Review by Kielek — APPROVED** [review]
LGTM, I do not see anything problematic if we speaking about Automatic Instrumentation.

Please remember to update release date in changelogs (can it be additional argument for creating PR?) when you decide to merge.

**Review by rajkumar-rangaraj — APPROVED** [review]

File: src/OpenTelemetry.Api.ProviderBuilderExtensions/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+## 1.9.0
+
+Released 2024-Jun-12
```
Comment (reyang) [inline src/OpenTelemetry.Api.ProviderBuilderExtensions/CHANGELOG.md:7]:
```suggestion
Released 2024-Jun-14
```

**Review by reyang — APPROVED** [review]
LGTM assuming the release date will be updated to Jun-14-2024.

**Review by alanwest — APPROVED** [review]

**Comment by CodeBlanch** [comment]
Opened a new one to get the correct dates: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5694

> Please remember to update release date in changelogs (can it be additional argument for creating PR?) when you decide to merge.

@Kielek I like the idea to make the date configurable. I'll jot it down and come to it at some point. Also thinking it would be good to mention who requested the release/kicked off the workflow.

**Pull request closed without merging**